Sources/_CryptoExtras/ChaCha20CTR/BoringSSL/ChaCha20CTR_boring.swift

```diff
@@ -0,0 +1,54 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+import Crypto
+@_implementationOnly import CryptoBoringWrapper
+import Foundation
+
+enum OpenSSLChaCha20CTRImpl {
+    static func encrypt<M: DataProtocol, N: DataProtocol>(key: SymmetricKey, message: M, counter: UInt32, nonce: N) throws -> [UInt8] {
+        guard key.bitCount == Insecure.ChaCha20CTR.keyBitsCount else {
+            throw CryptoKitError.incorrectKeySize
+        }
+        guard nonce.count == Insecure.ChaCha20CTR.nonceByteCount else {
+            throw CryptoKitError.incorrectParameterSize
+        }
+
+        let plaintext = Array(message)
+        var ciphertext = Array<UInt8>(repeating: 0, count: plaintext.count)
+        let nonce = Array<UInt8>(nonce)
+
+        self.chacha20CTR(out: &ciphertext, plaintext: plaintext, inLen: plaintext.count, key: key.withUnsafeBytes { Array($0) }, nonce: nonce, counter: counter)
```

Comment:
Rather than create a temporary Array, can you wrap this call in `key.withUnsafeBytes` and pass the closure argument as the `key` parameter?

Reply:
Hopefully https://github.com/apple/swift-crypto/pull/169/commits/1c7dcfc7b603f7b4ea4274ac2a6b5ad99c7f962e is what you had in mind? I tried to reference other calls into BoringSSL throughout the code to keep the style consistent.